### DISCUSSION TITLE
A question about using Trame with VTK9.2.6

### DISCUSSION THREAD

**Discussion by lkaly** [Kitware/trame#261] [Trame/VTK] [answered]
Hello, sorry to bother you. I have encountered an issue and I would appreciate some help. I am using Trame and VTK, and I noticed that the _solution_ray_cast.py_ file in the _01_vtk_ folder uses **vtkFixedPointVolumeRayCastMapper** for volume rendering of structured files. Now, I am trying to write a code for volume rendering of unstructured files using **vtkSmartVolumeMapper**.

![image](https://github.com/Kitware/trame/assets/59890189/e82f62e2-1dd6-488c-9bc2-589e6726edd8)
Figure 1.
![image](https://github.com/Kitware/trame/assets/59890189/0994939e-84b8-4dbc-9878-31885bcc72dc)
Figure 2.
I have consulted the official _VTK_ documentation, and both **vtkFixedPointVolumeRayCastMapper** and **vtkSmartVolumeMapper** are derived from **vtkVolumeMapper**(as Figure 3). In the _solution_ray_cast.py_ file, the command shown in Figure 4 is used. I attempted to use a similar command for **vtkSmartVolumeMapper** as Figure 5, but it resulted in an error as Figure 6. I'm wondering what could be the issue here?
![image](https://github.com/Kitware/trame/assets/59890189/033684ac-7b53-4449-aeb6-da622dab44cb)
Figure 3.
![image](https://github.com/Kitware/trame/assets/59890189/c97c2938-44c4-455a-a473-0238bebd65a2)
Figure 4.
![image](https://github.com/Kitware/trame/assets/59890189/674c34c0-700c-4d29-b8a5-fc6f2c51638f)
Figure 5.
![image](https://github.com/Kitware/trame/assets/59890189/985ed65d-9a8e-4fb1-9e0f-4b737ef2b7b3)
Figure 6.

If I want to find out the inheritance relationship of certain VTK classes in Python in the future, where can I look it up?
Thanks very much.

**Comment by jourdain** [comment, marked as answer]
A Google search should bring you to the doxygen documentation and class diagram but if you want to find in which vtk module it is available you will need to search in the code. 

The easier approach for you right now would be to do the following

```python
import vtk

mapper = vtk.vtkSmartVolumeMapper()
```

**Reply by lkaly** [reply]
Thanks very much!
